src/utility_belt/sql/conv.clj

```diff
@@ -1,61 +1,47 @@
 (ns utility-belt.sql.conv
-  (:require [clojure.java.jdbc :as jdbc]
+  (:require [next.jdbc.result-set :as jdbc.result-set]
+            [next.jdbc.prepare :as jdbc.prepare]
             [cheshire.core :as json]
             [clj-time.coerce :as coerce])
-  (:import (org.postgresql.util PGobject)
+  (:import (clojure.lang IPersistentMap IPersistentVector)
+           (org.postgresql.util PGobject)
+           (org.joda.time DateTime)
+           (java.util Date)
            (java.sql PreparedStatement)
            (org.postgresql.jdbc PgArray)))
 
-(extend-protocol jdbc/IResultSetReadColumn
-  org.postgresql.jdbc.PgArray
-  (result-set-read-column [val metadata _]
+(extend-protocol jdbc.result-set/ReadableColumn
+  PgArray
+  (result-read-column-by-index [val _meta _idx]
     (vec (.getArray val)))
-
-  ;; When reading JSONB, parse it out to clojure structure
   PGobject
-  (result-set-read-column [pgobj metadata idx]
-    (let [type  (.getType ^PGobject pgobj)
-          value (.getValue ^PGobject pgobj)]
+  (read-column-by-index [val _meta _idx]
+    (let [type  (.getType ^PGobject val)
+          value (.getValue ^PGobject val)]
       (case type
         "json" (json/parse-string value true)
         "jsonb" (json/parse-string value true)
-        value)))
-  ;; When reading timestamp, parse it out to java datetime
-  java.sql.Timestamp
-  (result-set-read-column [pgobj metadata idx]
-    (coerce/to-date-time pgobj)))
-
+        value))))
 
 (defn value-to-json-pgobject [value]
   (doto (PGobject.)
     (.setType "jsonb")
     (.setValue (json/generate-string value))))
 
-(extend-protocol jdbc/ISQLValue
-  ;; When inserting clojure structure convert it to JSONB
-  clojure.lang.IPersistentMap
-  (sql-value [value] (value-to-json-pgobject value))
-  clojure.lang.IPersistentVector
-  (sql-value [value] (value-to-json-pgobject value))
-  ;; When inserting datetime convert it to sql timestamp
-  org.joda.time.DateTime
-  (sql-value [value]
-    (coerce/to-sql-time value)))
-
-(extend-protocol jdbc/ISQLParameter
```

Comment:
@AfonsoTsukamoto  @MarketaAdamova I'm not sure what this was for - is it something about handling arrays coming out of `json_agg` results?

Reply:
so the jdbc/ISQLValue is needed so we can store data into pg columns without casting to correct data type.

I am not sure about jdbc/ISQLParameter, @AfonsoTsukamoto do you remember why we added this?

Reply:
The jdbc/ISQLValue is taken care of via the new jdbc.result-set/ReadableColumn protocol - tests confirm it's working as expected.
I'll extend the tests to account for more complicated types in JSONB column type and include them in vectors.

Reply:
@AfonsoTsukamoto ah perfect - that's what I was missing! I'll add a test 👍

Reply:
@AfonsoTsukamoto Added a test for inserting UUIDs https://github.com/nomnom-insights/nomnom.utility-belt.sql/pull/2/commits/8a69bb274692a0059cf70218014f9c77e15657cc - I've also added the same test in the `master` branch and results are the same 🙌

Reply:
(I didn't push the test to `master` as it would mess up the merge)